platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/stats/ReconnectMapMetrics.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.merkle.synchronization.stats;
+
+import com.swirlds.metrics.api.Counter;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+
+/**
+ * An implementation of ReconnectMapStats that emits all the stats as Counter metrics.
+ */
+public class ReconnectMapMetrics implements ReconnectMapStats {
+
+    private static final String RECONNECT_MAP_CATEGORY = "ReconnectMap";
+
+    private final ReconnectMapStats aggregateStats;
+
+    private final Counter transfersFromTeacher;
+    private final Counter transfersFromLearner;
+
+    private final Counter internalHash;
+    private final Counter internalCleanHash;
+    private final Counter internalData;
+    private final Counter internalCleanData;
+
+    private final Counter leafHash;
+    private final Counter leafCleanHash;
+    private final Counter leafData;
+    private final Counter leafCleanData;
+
+    /**
+     * Create an instance of ReconnectMapMetrics.
+     * @param metrics a non-null Metrics object
+     * @param label an optional label, e.g. a VirtualMap name or similar, may be null.
+     *     If specified, the label is added to the names of the metrics.
+     * @param aggregateStats an optional aggregateStats object to which all ReconnectMapStats calls will delegate
+     *     in addition to emitting metrics for this object, may be null.
+     */
+    public ReconnectMapMetrics(
+            @NonNull final Metrics metrics,
+            @Nullable final String label,
```

Comment:
I assume this class will be initialized and used somewhat similarly to `VirtualMapStatistics` or `MerkleDbStatistics`. If so, the label should be normalized, so all `.`s are replaced with `_`s

Reply:
Done.